dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUI.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include "pch.h"
+#include <future>
+#include <ppltasks.h>
+#include <pplawait.h>
+#include <ShlObj.h>
+#include "CameraCaptureUI.h"
+#include "Microsoft.Windows.Media.Capture.CameraCaptureUI.g.cpp"
+#include "CameraCaptureUITelemetry.h"
+#include "TelemetryHelper.h"
+
+//using namespace winrt::Windows::Media::Capture;
+using namespace winrt::Windows::UI;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::ApplicationModel::DataTransfer;
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    IAsyncOperation<hstring> LaunchCameraForResultToken(winrt::Microsoft::UI::WindowId const& targetWindow, ValueSet const& properties)
+    {
+        LauncherOptions options;
+        if (targetWindow.Value)
+        {
+            check_hresult(options.as<IInitializeWithWindow>()->Initialize(reinterpret_cast<HWND>(targetWindow.Value)));
+        }
+
+        options.TreatAsUntrusted(false);
+        options.DisplayApplicationPicker(false);
+        options.TargetApplicationPackageFamilyName(L"Microsoft.WindowsCamera_8wekyb3d8bbwe");
+
+        Uri launchUri{ L"microsoft.windows.camera.picker:" };
+        winrt::Windows::System::LaunchUriResult result = co_await Launcher::LaunchUriForResultsAsync(launchUri, options, properties);
+        if (!result.Result())
```

Comment:
Should we check results.Status() before this?

Reply:
We don't necessarily need to check result.Status() explicitly. If the operation is successful, we'll receive a file; in all other cases, we'll return null along with S_OK. This aligns with the UWP behavior, which handles similar scenarios gracefully without throwing exceptions.

UWP behavior which we are maintaining parity with - 
Successful Capture: Returns a storage object with S_OK.
Cancelled Capture: Returns null with S_OK. (No exceptions).
App Uninstalled: Displays a dialog box (Search for app in store) Yes/No? clicking on No returns null with S_OK.
Incorrect URI Mapping: Returns null with S_OK (No exceptions).

Reply:
What I meant is to use LaunchUriStatus to know if the app was launch successful or not **as it is done in original winrt API**. I didn't mention anywhere that you need to necessarily throw exception in-case of failure to launch. You can simply log the case and return empty string to handle it gracefully and this is a suggestion only, feel free to skip it in this iteration.